tests/rptest/services/redpanda.py

```diff
@@ -3346,6 +3687,9 @@ def validate_controller_log(self):
                 )
 
 
-def make_redpanda_service(environment):
+def make_redpanda_service(context, num_brokers, **kwargs):
```

Comment:
shouldn't this be added to all existing tests?  i wonder if there's a way to inject this factory as part of the invocation to the parent class. Though that would be kind of hacky

Reply:
i was initially being conservative in this PR by just using the factory function in `CloudSelfTest`, but i'll change it to replace everywhere. i counted 17 files that get changed which is not crazy and the commit is a simple replace.

i think the factory function fits in well here since its signature is the same as the constructor of `RedpandaService`.